### PR TITLE
feat: Simple transaction router

### PR DESCRIPTION
**Pull request by alaibe**
* Add a new transaction endpoint accepting the chain id as a parameter
* Add a simple router, which check amongst all networks available the one where a transaction can be routed to

### CODE REVIEW

File: api/geth_backend.go [hunk 1]
```diff
@@ -874,6 +874,22 @@ func (b *GethStatusBackend) SendTransaction(sendArgs transactions.SendTxArgs, pa
...
+	}
+
+	hash, err = b.transactor.SendTransactionWithChainID(chainID, sendArgs, verifiedAccount)
+	if err != nil {
+		return
+	}
+
+	go b.statusNode.RPCFiltersService().TriggerTransactionSentToUpstreamEvent(hash)
+
+	return
```
Comment (saledjenic) [inline api/geth_backend.go:891]: @alaibe is this needed here?

File: services/wallet/router.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+	}
+
+	var (
+		group      = async.NewAtomicGroup(ctx)
+		mu         sync.Mutex
+		candidates = make([]params.Network, 0)
+	)
+	for networkIdx := range networks {
+		network := networks[networkIdx]
+		if network.IsTest != areTestNetworksEnabled {
```
Comment (saledjenic) [inline services/wallet/router.go:70]: @alaibe is this condition what you really want to achieve? 
That condition means you're adding network in the following 2 cases:
- if it is a test network and test networks are enabled 
- if it is not a test network and test networks are not enabled 

but I think you should add it in the following case as well:
- if it is not a test network and test networks are enabled

if above is true then condition should be like:
```
if network.IsTest && !areTestNetworksEnabled {
    continue
}
```

File: api/geth_backend.go (hunk 1 above)
Comment (alaibe) [reply api/geth_backend.go:891]: Will double check and remove if not needed

File: services/wallet/router.go (hunk 2 above)
Comment (alaibe) [reply services/wallet/router.go:70]: > if it is not a test network and test networks are enabled

Why do think I should add this:
I don t want to make transaction on mainnet if test networks are enable.
If test networks are enable, I only want to transfer on ropsten/rinkeby...

File: api/geth_backend.go (hunk 1 above)
Comment (richard-ramos) [reply api/geth_backend.go:891]: It's needed when you use named return parameters

File: services/wallet/router.go [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+type Router struct {
+	s *Service
+}
+
+func (r *Router) suitableTokenExists(ctx context.Context, network *params.Network, tokens []*Token, account common.Address, amount float64, tokenSymbol string) (bool, error) {
+	for _, token := range tokens {
+		if token.Symbol != tokenSymbol {
+			continue
+		}
```
Comment (richard-ramos) [inline services/wallet/router.go:31]: What do you think of checking the token address instead of symbol?

Comment (richard-ramos) [reply services/wallet/router.go:31]: Ah no, this is supposed to be multi chain, right? so the addresses are gonna be different.

**Review by richard-ramos — APPROVED** [review]

File: services/wallet/router.go (hunk 3 above)
Comment (alaibe) [reply services/wallet/router.go:31]: correct, I need to use the symbol as key

**Review by Samyoul — APPROVED** [review]

**Pull request merged**